Neos.Flow/Classes/Mvc/DispatchMiddleware.php

```diff
@@ -0,0 +1,90 @@
+<?php
+namespace Neos\Flow\Mvc;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Http\Component\ComponentContext;
+use Neos\Flow\Mvc\Routing\RoutingComponent;
+use Neos\Flow\Security\Context;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * A MVC dispatch middleware
+ *
+ * Note, this is a final middleware that will not delegate further.
+ */
+class DispatchMiddleware implements MiddlewareInterface
```

Comment:
Not yet used - but this is how the dispatch component could look as a middleware